packages/design-system/src/components/Dialog/Dialog.jsx

```diff
@@ -169,6 +185,11 @@ Dialog.propTypes = {
    * The Dialog's heading, to be rendered in the header alongside the close button.
    */
   heading: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  /**
+   * Set focus to a specific element that should receive initial focus (if `focusDialog={false}`).
+   * [Read more on react-aria-modal docs.](https://github.com/davidtheclark/react-aria-modal#initialfocus)
+   */
+  initialFocus: PropTypes.string,
```

Comment:
That's right, we don't use this prop on `<Dialog>` because when the modal is opened, by default, `<Dialog>` would place the focus on the entire modal (we call this out under Focus management). 
If applications want to place the focus inside the modal, the inherent `<AriaModal>` props can be passed through to dialog. Eg, to place focus inside the dialog when the modal is opened, they can do this by:
- passing [`initialFocus` set to the selector string for focus](https://github.com/davidtheclark/react-aria-modal#initialfocus), and 
- passing `focusDialog` boolean flag set to  `false`

Reply:
We hope that by adding this prop `initialFocus` on Dialog, it will call attention to it for users to use whenever they wish to place focus on the inside of modal.